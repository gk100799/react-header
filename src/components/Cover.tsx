import React, { Component } from 'react'

import { CoverProps } from './Constants'

export default class Cover extends Component<CoverProps> {

    render() {
        return (
            <div className="d-flex justify-content-center align-items-center h-100 w-100">
                <main role="main" className="inner cover">
                    <h1 className="cover-heading" data-testid="coverHeading">{this.props.coverHeading}</h1>
                    <p className="lead" data-testid="coverContent">{this.props.coverContent}</p>
                    <p className="lead">
                        <button data-testid="button" className="btn btn-lg btn-secondary">{this.props.buttonTitle}</button>
                    </p>
                </main>
            </div>
        )
    }
}
