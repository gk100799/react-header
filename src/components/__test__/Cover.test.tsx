import React from 'react';
import ReactDOM from 'react-dom'
import Cover from '../Cover'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

afterEach(cleanup);

it("renders Cover without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(<Cover />, div)
})

it("renders Cover correctly", () => {
    const props = {
        coverHeading: 'This is Home page.',
        coverContent: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Totam magni illum molestiae reiciendis et quis libero porro neque suscipit beatae sunt facilis ullam corrupti temporibus voluptatibus, facere, quod corporis similique?',
        buttonTitle: 'Subscribe'
    }
    const { getByTestId } = render(<Cover {...props} />)
    expect(getByTestId("coverHeading")).toHaveTextContent(props.coverHeading)
    expect(getByTestId("coverContent")).toHaveTextContent(props.coverContent)
    expect(getByTestId("button")).toHaveTextContent(props.buttonTitle)
})

it("matches snapshot", () => {
    const props = {
        coverHeading: 'This is Home page.',
        coverContent: ' Lorem ipsum dolor sit amet consectetur adipisicing elit. Totam magni illum molestiae reiciendis et quis libero porro neque suscipit beatae sunt facilis ullam corrupti temporibus voluptatibus, facere, quod corporis similique? ',
        buttonTitle: 'Subscribe'
    }
    const tree = renderer.create(<Cover {...props} />)
    expect(tree).toMatchSnapshot();
})

