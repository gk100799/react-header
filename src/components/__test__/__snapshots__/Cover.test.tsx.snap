// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="d-flex justify-content-center align-items-center h-100 w-100"
>
  <main
    className="inner cover"
    role="main"
  >
    <h1
      className="cover-heading"
      data-testid="coverHeading"
    >
      This is Home page.
    </h1>
    <p
      className="lead"
      data-testid="coverContent"
    >
       Lorem ipsum dolor sit amet consectetur adipisicing elit. Totam magni illum molestiae reiciendis et quis libero porro neque suscipit beatae sunt facilis ullam corrupti temporibus voluptatibus, facere, quod corporis similique? 
    </p>
    <p
      className="lead"
    >
      <button
        className="btn btn-lg btn-secondary"
        data-testid="button"
      >
        Subscribe
      </button>
    </p>
  </main>
</div>
`;
